src/main.ts

```diff
@@ -0,0 +1,528 @@
+import * as actions_core from "@actions/core";
+import { mkdtemp, chmod, access } from "node:fs/promises";
+import { spawn } from "node:child_process";
+import { join } from "node:path";
+import { tmpdir } from "node:os";
+import { pipeline } from "node:stream";
+import fetch from "node-fetch";
+import { promisify } from "node:util";
+import fs from "node:fs";
+import stringArgv from "string-argv";
+
+class NixInstallerAction {
+  platform: string;
+  nix_package_url: string | null;
+  backtrace: string | null;
+  extra_args: string | null;
+  extra_conf: string[] | null;
+  github_token: string | null;
+  // TODO: linux_init
```

Comment:
Do we want to do this now or later?

Reply:
Might as well

Reply:
Thinking on this more, I actually don't know if we should do this. While MacOS might not have an init setting (since we need the services for volume mounting), a future OS we add, such as FreeBSD, will likely have this option.